### ISSUE TITLE
[Feature request] Omit mapped type

### ISSUE THREAD

**Issue by abenhamdine** [Microsoft/TypeScript#13980]
First, thx a lot for the fantastic stuff 😄 

Since ts 2.1, we have mapped types that are very usefull.
_Pick_ is now included in the library, and allow to pick only some properties of a type T.
While Pick is usefull, it would be great to construct a type base on another type by **omitting** some properties, ie the opposite of _Pick_

 Notably, it would allow to override some interfaces, eg : 

```ts
interface Foo {
    name : string,
    age: number,
    id : string
}

// now we want the id as a number in a particular piece of code, along with the other properties
interface DifferentFoo extends Omit<Foo, "id"> {
   id: number
}

**Comment by aluanhaddad** [comment]
Great idea but also likely a duplicate of #12424

**Comment by abenhamdine** [comment]
Hmm.. I didn't read very carefully #12424 (it's very dense...), but is this exactly the same request ?

#12424 seems to ask for a mapping  to selected kinds of types, while I ask for a conditionnal mapping based on property names.

**Comment by abenhamdine** [comment]
I rather think this is the same as #12215 (omit word is even already mentionned)

**Comment by abenhamdine** [comment]
closed in favor of #12215, my example doesn't add any significant value to it.
